src/ai/defaultai.cc

```diff
@@ -1297,125 +1297,130 @@ void DefaultAI::late_initialization() {
 /**
  * Checks PART of available buildable fields.
  *
- * this checks about 40-50 buildable fields. In big games, the player can have thousands
- * of them, so we rotate the buildable_fields container and check 35 fields, and in addition
- * we look for medium & big fields and near border fields if needed.
+ * It has 3 stages:
+ * 1. Checking for invalid fields (not more buildable, e.g. tree grew there and so on) and updating
+ *  specific types of fields to the limit as quantified in special_fields_to_preffer array (no
+ *  rotating of buildable_fields here)
+ * 2. Removing invalid fields and partial rotating of the deque as a side effect
+ * 3. Updating further BFs up to the overall limit of max_fields_to_check (no rotating)
+ * As a rule we update fields with expired info only.
+ * The total count of BF to be updated is hardlimited to max_fields_to_check (30 now)
  */
 void DefaultAI::update_all_buildable_fields(const Time& gametime) {
 
-	// Every call we try to check first 35 buildable fields
-	constexpr uint16_t kMinimalFieldsCheck = 35;
+	if (buildable_fields.empty()) {
+		return;
+	}
 
-	uint16_t i = 0;
+	// This defines count of preffered types of fields to update info stright away
+	uint16_t special_fields_to_preffer[3] = {6, 6, 5};
+	// Positions and types are defined here
+	const uint16_t kSpecialFieldPos = 0;
+	const uint16_t kMediumlFieldPos = 1;
+	const uint16_t kBigFieldPos = 2;
+	const uint16_t kNoReasonPos = 3;
+
+	// The overall limit should be of course higher than the sum of above special fields
+	const u_int16_t max_fields_to_check = 30;
```

Comment:
this is a typo that prevents compiling on my machine

Reply:
strange. Fixed